tsinfer/formats.py

```diff
@@ -2480,6 +2480,127 @@ def insert_proxy_samples(
         assert other.num_ancestors == self.num_ancestors + len(sample_ids)
         return other
 
+    def truncate_ancestors(
+        self,
+        lower_time_bound,
+        upper_time_bound,
+        length_multiplier=2,
+        **kwargs,
+    ):
+        """
+        Truncates the length of ancestors above a given time and returns a new
+        :class:`.AncestorData` instance.
+
+        The maximum length of an ancestor existing between ``lower_time_bound'' and
+        ``upper_time_bound'', multiplied by ``length_multplier'', is used to truncate
+        the length of all ancestors older than ``upper_time_bound'' which contain
+        at least one ``focal_site''.
+
+        Specifically, the haplotypes of ancestors older than ``upper_time_bound'' may
+        extend no further than *half of this maximum length* to the either side of the
+        leftmost and rightmost focal sites on each ancestor. Note that ancestors above
+        ``upper_time_bound'' may still be longer than the maximum length if the
+        ancestor contains greater than 2 ``focal_sites''.
+
+        This function should be used when :func:`tsinfer.generate_ancestors` generates
+        old ancestors which are very long, as this can adversely affect performance.
```

Comment:
"performance" could be interepreted as "statistical performance". Better to be concrete "as this can significantly slow down matching time".